insights/combiners/ipa.py

```diff
@@ -0,0 +1,73 @@
+"""
+IPA - Combiner for RHEL IdM / FreeIPA information
+=================================================
+"""
+from insights.core.plugins import combiner
+from insights.core.exceptions import SkipComponent
+from insights.parsers.installed_rpms import InstalledRpms
+from insights.parsers.redhat_release import RedhatRelease
+from insights.parsers.ipa_conf import IPAConfig
+from insights.parsers.sssd_conf import SSSD_Config
+
+
+@combiner(IPAConfig, SSSD_Config, InstalledRpms, RedhatRelease)
+class IPA(object):
+    """Combiner for IPA, SSSD, and installed RPMs
+
+    Provides additional information, e.g. whether the host is an IPA server.
+    """
+
+    def __init__(self, ipa_conf, sssd_conf, rpms, release):
+        self._ipa_conf = ipa_conf
+        self._sssd_conf = sssd_conf
+        # IPA package names are different on Fedora
+        if release.is_fedora:
+            self._client_rpm = rpms.get_max("freeipa-client")
+            self._server_rpm = rpms.get_max("freeipa-server")
+        else:
+            self._client_rpm = rpms.get_max("ipa-client")
+            self._server_rpm = rpms.get_max("ipa-server")
+        if self._client_rpm is None:
+            raise SkipComponent("IPA client package is not installed")
+
+    @property
+    def ipa_conf(self):
+        """Get IPAConfig object"""
+        return self._ipa_conf
+
+    @property
+    def sssd_conf(self):
+        """Get SSSD_Config object"""
+        return self._sssd_conf
+
+    @property
+    def sssd_domain_config(self):
+        """Get SSSD domain configuration for host's IPA domain"""
+        return self._sssd_conf.domain_config(self._ipa_conf.domain)
+
+    @property
+    def is_client(self):
+        """Is the host an IPA client?"""
+        if not self._client_rpm:
+            return False
+        cfg = self.sssd_domain_config
+        if not cfg.get("id_provider") == "ipa":
+            return False
+        return True
+
+    @property
+    def is_server(self):
```

Comment:
All other `property`s look good to me, however, this one (and the `is_client`) looks a bit heavy, could you mind introducing one more internal member to do the calculation in advance (e.g. in the `__init__`)?  (Since the insights-core needs to support various py versions, it's not so flexible to allow to use the `cache_property`)

Reply:
The `is_client` and `is_server` properties aren't doing any computationally heavy workload. It's just a bunch of boolean checks, dict lookups with string comparison and one `str.startswith()`. If you like I can add manual caches to the properties.